webcompat/static/js/lib/bugform.js

```diff
@@ -573,6 +605,136 @@ function BugForm() {
+
+    // TODO ask Mike if there is a better way to share this globally, like this.ctx
+    globalCtx = ctx;
+    globalCtx.drawImage(img, 0, 0);
+
+    // TODO ask Mike if there is a better way
+    // having to define the canvas event listeners here because at time
+    // of init definition, the canvas does not exist
+    var domCanvas = $("#mCanvas");
+    domCanvas.on("mousedown", _.bind(this.canvasMouseDown, this));
+    domCanvas.on("mousemove", _.bind(this.canvasMouseDrag, this));
+    domCanvas.on("mouseup", _.bind(this.canvasMouseUp, this));
+
+  }
+
+  this.closeImageEditor = function() {
+    var myCanvas = document.querySelector('canvas');
+ 
+    this.dataURI = myCanvas.toDataURL();
+    this.editImageModal.addClass("is-hidden");
+    this.addPreviewBackground(this.dataURI);
+  }
+
+  this.setImageEditorHighlight = function() {
+    this.curEditorActiveControl = this.ACTIVE_CONTROL_OPTIONS[0];
+  }
+
+  this.setImageEditorRedact = function() {
+    this.curEditorActiveControl = this.ACTIVE_CONTROL_OPTIONS[1];
+  }
+
+  this.canvasMouseDown = function(ev) {
+    var myCanvas = document.querySelector('canvas');
+    var mPos = this.getMousePos(myCanvas, ev);
+
+    this.initMouseDownPos = mPos;
+    this.mouseDown = true;      
+  }
+
+  this.canvasMouseDrag = function(ev) {
+    var myCanvas = document.querySelector('canvas');
+    var mPos = this.getMousePos(myCanvas, ev);
+
+    // drawing lines for shape border
+    // 1 need to draw the first 5px line separately
+    if (this.mouseDown && mPos.x <= this.initMouseDownPos.x - 5 || mPos.x >= this.initMouseDownPos.x + 5 || mPos.y <= this.initMouseDownPos.y - 5 || mPos.y >= this.initMouseDownPos.y + 5) {
+      this.mouseClickDragActive = true;
+
+      if (this.curEditorActiveControl === this.ACTIVE_CONTROL_OPTIONS[0]) {
+        globalCtx.lineWidth = 2;
+        globalCtx.strokeStyle =  '#ffff00';
+      } else {
+        globalCtx.lineWidth = 1;
+        globalCtx.strokeStyle =  '#000';
+      }
+
+      globalCtx.beginPath();
+      globalCtx.moveTo(this.initMouseDownPos.x, this.initMouseDownPos.y);
+      globalCtx.lineTo(this.initMouseDownPos.x, mPos.y);
+      globalCtx.stroke();
+
+      globalCtx.beginPath();
+      globalCtx.moveTo(this.initMouseDownPos.x, this.initMouseDownPos.y);
+      globalCtx.lineTo(mPos.x, this.initMouseDownPos.y);
+      globalCtx.stroke();
+
+    }
+
+    // 2 drawing the shape border for mouse position diff > 5px
+    //if (mouseDown && mPos.x < initMouseDownPos.x - 5 || mPos.x > initMouseDownPos.x + 5 || mPos.y < initMouseDownPos.y - 5 || mPos.y > initMouseDownPos.y + 5 && mouseClickDragActive) {
+        // console.log('need to continue drawing lines for border here');
+    //}
+
+  }
+
+  this.canvasMouseUp = function(ev) {
+    var myCanvas = document.querySelector('canvas');
+    var mPos = this.getMousePos(myCanvas, ev);
+
+    // only do this if event was not mouseup on control button
+    if (mPos.y >= 160) {
```

Comment:
my bad, this was something I should have deleted from the original demo code... done.